src/gpu/intel/ocl/stream.hpp

```diff
@@ -116,8 +121,14 @@ struct stream_t : public compute::compute_stream_t {
             cl_context ctx, cl_device_id dev, cl_int *err) const;
 
     std::unique_ptr<mdapi_helper_t> mdapi_helper_;
+
+    cl_event async_tracked_event_ = nullptr;
+    async_timing_data_t *stream_timing_data_ = nullptr;
```

Comment:
Please use smart pointers to make sure it's properly destroyed in case of an early return (e.g. because of some error).

```suggestion
    std::unique_ptr<async_timing_data_t> stream_timing_data_;
```

Edit: I see that you used `shared_ptr` in SYCL, let's align OpenCL with SYCL. It also seems `unique_ptr` would be enough.